apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/WitnessClassFinder.java

```diff
@@ -0,0 +1,42 @@
+package org.skywalking.apm.agent.core.plugin;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import net.bytebuddy.pool.TypePool;
+
+/**
+ * The <code>WitnessClassFinder</code> represents a pool of {@link TypePool}s,
+ * each {@link TypePool} matches a {@link ClassLoader},
+ * which helps to find the class define existed or not.
+ *
+ * @author wusheng
+ */
+public enum WitnessClassFinder {
+    INSTANCE;
+
+    private Map<ClassLoader, TypePool> poolMap = new ConcurrentHashMap<ClassLoader, TypePool>();
+
+    /**
+     * @param witnessClass
+     * @param classLoader for finding the witnessClass
+     * @return true, if the given witnessClass exists, through the given classLoader.
+     */
+    public boolean exist(String witnessClass, ClassLoader classLoader) {
+        ClassLoader mappingKey = classLoader == null ? NullClassLoader.INSTANCE : classLoader;
+        if (!poolMap.containsKey(witnessClass)) {
```

Comment:
Yes, the key should be `classLoader`. By using double locks, I think we should change the `ConcurrentHashMap` to `HashMap` too.